eox_tagging/test/test_models.py

```diff
@@ -0,0 +1,183 @@
+        self.fake_object_tagged = CourseFakeModel.objects.create()  # pylint: disable=no-member
+        self.fake_object_tagged_fail = EnrollmentsFakeModel.objects.create()  # pylint: disable=no-member
+
+        self.test_tag = Tag.objects.create(
+            tag_value="testValue",
+            tag_type="testType",
+            tagged_object=self.tagged_object,
+            belongs_to=self.belongs_to_object,
+        )
+
+        Tag.objects.create(
+            tag_value="testValue",
+            tag_type="testType",
+            tagged_object=self.fake_object_tagged,
+            belongs_to=self.fake_belongs_to_object,
+        )
+
+    @override_settings(EOX_TAGGING_DEFINITIONS=[])
+    def test_empty_setting(self):
+        """ Used to test saving without validations defined"""
+        Tag.objects.create(
+            tag_value="testValue",
+            tag_type="testType",
+            tagged_object=self.fake_object_tagged,
+            belongs_to=self.fake_belongs_to_object,
+        )
+
+    def test_valid_tag(self):
+        """ Used to confirm that the tags created are valid."""
+        tag_status = getattr(self.test_tag, "status")
+
+        self.assertEqual(tag_status, 1)
+
+    def test_tag_value(self):
+        """ Used to confirm that the tag_value is correct."""
+        tag_value = getattr(self.test_tag, "tag_value")
+
+        self.assertEqual(tag_value, "testValue")
+
+    def test_tag_type(self):
+        """ Used to confirm that the tag_type is correct."""
+        tag_value = getattr(self.test_tag, "tag_type")
+
+        self.assertEqual(tag_value, "testType")
+
+    def test_tag_value_not_in_settings(self):
+        """ Used to confirm validation error when the value is not defined."""
+        with self.assertRaises(ValidationError):
+            Tag.objects.create(
+                tag_value="testValues",
+                tag_type="testType",
+                tagged_object=self.tagged_object,
+                belongs_to=self.belongs_to_object,
+            )
+
+    def test_tag_type_not_in_settings(self):
+        """ Used to confirm validation error when the type is not defined."""
+        with self.assertRaises(ValidationError):
+            Tag.objects.create(
+                tag_value="testValue",
+                tag_type="testTypes",
+                tagged_object=self.tagged_object,
+                belongs_to=self.belongs_to_object,
+            )
+
+    def test_tag_different_generic_objects_fail(self):
+        """ Used to confirm the tags can't be use with objects not defined in settings."""
+        with self.assertRaises(ValidationError):
+            Tag.objects.create(
+                tag_value="testValue",
+                tag_type="testType",
+                tagged_object=self.fake_object_tagged_fail,
+                belongs_to=self.belongs_to_object,
+            )
+
+    def test_tag_different_generic_objects(self):
```

Comment:
same comment line 46

Reply:
What we want to test is we can tag multiple objects, we are testing the generic part of the model. 
The creation must succeed because there is not a restriction in the objects we can tag.
Do you think we can add more to the test?